Plugins/Traccc/include/Acts/Plugins/Traccc/TrackConversion.hpp

```diff
@@ -0,0 +1,271 @@
+/// @param e the traccc container element.
+/// @return the fitting result contained in the container element (i.e., e.header).
+template <typename fitting_result_t, typename track_state_vector_t>
+inline auto& getFittingResult(
+    const traccc::container_element<fitting_result_t, track_state_vector_t>&
+        tracccTrack) {
+  return tracccTrack.header;
+}
+
+/// @brief Gets the track states of a traccc container element.
+/// @param e the traccc container element.
+/// @return the track states contained in the container element (i.e., e.items).
+template <typename fitting_result_t, typename track_state_vector_t>
+inline auto& getTrackStates(
+    const traccc::container_element<fitting_result_t, track_state_vector_t>&
+        tracccTrack) {
+  return tracccTrack.items;
+}
+
+/// @brief Creates a new track in the Acts track container.
+/// This new track will contain data copied from the traccc track container
+/// element (track and track state data).
+/// @param tracccTrack The traccc container element to copy from.
+/// @param actsTrackContainer The Acts track container. This is the new track will be made in this container.
+/// @param detector The detray detector.
+/// @param trackingGeometry The Acts tracking geometry.
+/// @note Does not set the uncalibrated source link and calibrated measurements.
+template <typename fitting_result_t, typename track_state_vector_t,
+          typename track_container_t, typename trajectory_t,
+          template <typename> class holder_t, typename metadata_t,
+          typename container_t>
+inline auto makeTrack(
+    const traccc::container_element<fitting_result_t, track_state_vector_t>&
+        tracccTrack,
+    Acts::TrackContainer<track_container_t, trajectory_t, holder_t>&
+        actsTrackContainer,
+    const detray::detector<metadata_t, container_t>& detector,
+    const Acts::TrackingGeometry& trackingGeometry) {
+  auto fittingResult = getFittingResult(tracccTrack);
+  auto trackStates = getTrackStates(tracccTrack);
+
+  auto track = actsTrackContainer.makeTrack();
+  copyFittingResult(fittingResult, track, detector, trackingGeometry);
+
+  // Make the track states.
+  for (const auto& tstate : trackStates) {
+    auto astate = track.appendTrackState();
+    copyTrackState(tstate, astate, detector, trackingGeometry);
+  }
+
+  track.linkForward();
+
+  return track;
+}
+
+/// @brief Creates a vector of Acts and traccc track state pairs.
+/// The states in the pairs are paired 1:1 by index.
+/// @tparam traccc_track_t type of traccc container_element.
+/// @tparam acts_track_t type of Acts track proxy.
+/// @param tracccTrack the traccc track.
+/// @param actsTrack the Acts track.
+/// @returns a vector of tuples.
+/// @note The Acts track and traccc track must contain the same number of track states.
+template <typename traccc_track_t, typename acts_track_t>
+auto trackStateZipView(traccc_track_t& tracccTrack, acts_track_t& actsTrack) {
```

Comment:
We should be careful with this! Can you elaborate what the use case is? The ACTS track EDM is designed specifically not to be a vector of track states.